File: Projet_Gestion_RDV-Stage_Final/App_Web_EL BAROUDI/CHUO/Classes/Medecin.php
<?php

    class Medecin extends User {

        //ATTRIBUTES
        protected $specialite;
        protected $NuMedecin;
        protected $conge;
        //CONSTRUCTORS

        public function __construct()
        {
            parent::__construct();
            $this->setDescription('Medecin');
            $this->specialite = null;
            $this->NuMedecin = null;
            $this->conge = null;


        }

        //SETTERS
        public function setSpecialite($specialite) : void { $this->specialite = $specialite; }
        public function setNuMedecin($NuMedecin) : void { $this->NuMedecin = $NuMedecin; }
        public function setConge($conge) : void { $this->conge = $conge; }

        //GETTERS
        public function getSpecialite() : string { return $this->specialite; }
        public function getNuMedecin() : string { return $this->NuMedecin; }
        public function getConge() : string { return $this->conge; }





    }

?>

File: Projet_Gestion_RDV-Stage_Final/App_Android_EL BAROUDI/chuo_Android/app/src/main/java/com/example/chuo_android/Medecin.java
package com.example.chuo_android;

public class Medecin extends User {
    private String specialite,numedecin;
    private static int taille=0;

    public Medecin(String cin, String nom, String prenom, String dateN, String telephone, String email, String password, String description, String specialite, String numedecin) {
        super(cin, nom, prenom, dateN, telephone, email, password,"Medecin");
        this.specialite = specialite;
        this.numedecin = numedecin;
        this.taille=0;
    }
    public Medecin(String nom,String prenom,String numedecin,int taille) {
        super(null, nom, prenom, null, null, null, null,"Medecin");
        this.specialite = null;
        this.numedecin = numedecin;
        this.taille=taille;

    }
    /*public Medecin(String nom,String prenom) {
        super(null, nom, prenom, null, null, null, null,"Medecin");
        this.specialite = null;
        this.numedecin = null;
        this.taille=0;

    }*/
    //setters
    public void setSpecialite(String specialite){this.specialite=specialite;}
    public void setNumedecin(String numedecin){this.numedecin=numedecin;}
    public void setTaille(int taille){this.taille=taille;}

    //getters
    public String getSpecialite() {
        return specialite;
    }
    public String getNumedecin() {
        return numedecin;
    }
    public int getTaille() {
        return taille;
    }
}
